### ISSUE TITLE
False Negative: UCF_USELESS_CONTROL_FLOW missed in catch blocks and simple terminal statements

### ISSUE THREAD

**Issue by leemeii** [spotbugs/spotbugs#3894]
**Describe the bug**
The rule **UCF_USELESS_CONTROL_FLOW**  fails to flag empty if statements located within catch blocks and ignores simple empty if blocks at the end of a method.

**To Reproduce**
```java
public class Bug {
    static int x = 0;
    static int y = 0;
  //case 1
    public static void test() {
        if (x > 0) {} //  <- triggered

 //case 2
        try { trigger(); } 
        catch (Exception e) {
            if (x == 0) {} // <- should report (FN)
        }
  //case 3
        if (y != 0) {} // <- should report (FN)
  //case 4
        if (x == 0 && true) {} // <-triggered 
    }
    static void trigger() throws Exception { throw new Exception(); }
}

```
Command: spotbugs -textui -low -effort:max Bug.class

**Expected behavior**
All four empty if blocks should be flagged as UCF violations regardless of their position or logical redundancy.

**Actual behavior**
Only  Case1 and Case4 are reported.

**Environment**
- Java version: JDK 17
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI
